### PR TITLE
Add GSoC 2024 page

### PR DESCRIPTION
**Pull request by XVilka**

### CODE REVIEW

File: content/gsoc/2024.md [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+- Megabeets
+
+### Links/Resources
+- [Issue #1041](https://github.com/rizinorg/cutter/issues/1041)
+- [Heap Viewer plugin for IDA Pro](https://github.com/danigargu/heap-viewer)
+- [Heap parsing for MacOS, tmalloc, jmalloc](https://github.com/rizinorg/rizin/issues/208)
+- [Dynamic Allocator Detection](https://github.com/rizinorg/rizin/issues/157)
+- ["heap"-marked Rizin issues](https://github.com/rizinorg/rizin/labels/heap)
+
+## Diffing mode (175 hour project)
```
Comment (wargio) [inline content/gsoc/2024.md:223]: this is going to be implemented by me. we can remove it

File: content/gsoc/2024.md [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+- xvilka
+- thestr4ng3r
+
+### Links/Resources
+- [RZIL-labeled issues](https://github.com/rizinorg/rizin/labels/RZIL)
+- [ESIL-labeled issues](https://github.com/rizinorg/rizin/labels/RZIL)
+- [ESIL to RZIL conversion tracking issue](https://github.com/rizinorg/rizin/issues/2080)
+- [Cutter: IL output and graph visual representation](https://github.com/rizinorg/rizin/issues/1133)
+
+## FRIDA Integration
```
Comment (wargio) [inline content/gsoc/2024.md:326]: there are two `Frida integration`. we should change the name of one of these.

File: content/gsoc/2024.md [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+- [Materials](http://dustri.org/b/files/hacklu2014_r2_exploitation.tar.xz) about the exloitation workshop at Hack.lu 2014
+- [Slides](https://github.com/XVilka/hacklu) for the exploitation part of workshop at Hack.lu 2015
+- [RzEgg related bugs](https://github.com/rizinorg/rizin/labels/RzEgg)
+
+# Microtasks
+
+When taking any of microtasks please be sure someone isn't already working on them, and let
+us know if you are going to work on a particular one.
+
+## Cutter
```
Comment (wargio) [inline content/gsoc/2024.md:505]:
```suggestion
## Cutter UX improvements
```

File: content/gsoc/2024.md [hunk 4]
```diff
@@ -0,0 +1,606 @@
...
+It should be possible to call this command with a specific class, so it only looks into its vtable, or without a class, so it gives a list of possible destinations for all vtables that are not too small for the offset.
+
+When that is implemented, one could also add a command that does the same thing, but automatically takes the offset from the opcode at the current seek.
+
+### Add classes list to `Vb`
+
+`Vb` already supports browsing bin classes. The same thing should be implemented for classes from
+analysis.
+
+## Refactoring
```
Comment (wargio) [inline content/gsoc/2024.md:557]:
```suggestion
## RIzin legacy code refactoring
```

File: content/gsoc/2024.md [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+
+## File formats
+
+Implementing the support for any new file format counts as a microtask. See [New File-Format](https://github.com/rizinorg/ideas/labels/New%20File%20Format) label for pending issues.
+
+## Disassemblers and assemblers
+
+Implementing the support for any new architecture counts as a microtask. See [New-Architecture](https://github.com/rizinorg/ideas/labels/New%20Architecture) label for pending issues.
+
+## Analysis
```
Comment (wargio) [inline content/gsoc/2024.md:527]:
```suggestion
## Rizin analysis improvements
```

File: content/gsoc/2024.md [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+- [User Experience project for Cutter](https://github.com/orgs/rizinorg/project/4/views/1)
+- [User Experience project for Rizin](https://github.com/rizinorg/rizin/projects/1)
+
+## Plugins and Python High Level API (175 hour project)
+
+Our current public API to be used by plugin authors is somewhat limited. We need to improve a lot of things about our Plugins support and take it few steps ahead.
+This task is only about improving the public C++ and Python interface of Cutter, specifically its graphical user interface components. For a task about exposing Rizin's API for disassembly, analysis and other purposes, see the Rizin bindings task above.
+
+### Task
+ - Expose everything Cutter can offer for plugins authors. This includes high level API, integration of the plugin management etc.
```
Comment (Rot127) [inline content/gsoc/2024.md:120]:
```suggestion
 - Expose everything Cutter can offer for plugin authors. This includes high level API, integration of the plugin management etc.
```

File: content/gsoc/2024.md [hunk 7]
```diff
@@ -0,0 +1,606 @@
...
+- [Update binutils code to latest](https://github.com/rizinorg/rizin/issues/814)
+
+## Exploitation capabilities improvements (175 hour project)
+
+Since modern architectures are now enforcing [W^X](https://en.wikipedia.org/wiki/W%5EX), exploiters are using [ROP](https://en.wikipedia.org/wiki/Return-oriented_programming). (Un)fortunately, building ROP chain by hand can be tedious, this is why some tools can be used to ease this construction: ImmunityDBG has [mona.py](https://www.corelan.be/index.php/2012/12/31/jingle-bofs-jingle-rops-sploiting-all-the-things-with-mona-v2/), there is also [ROPgadget](http://www.shell-storm.org/project/ROPgadget/) and [dropper](https://github.com/rizlik/dropper).There exist even tools that can generate ROP chains automatically, for example [exrop](https://github.com/d4em0n/exrop). It's a shame that despite having [ESIL](https://book.rizin.re/disassembling/esil.html), Rizin doesn't have something similar yet. One of the possible solutions would be to build an external plugin or tool which will reuse power of librz and rz-gg. Moreover it makes sense to think about SROP, COOP and BROP support.
```
Comment (Rot127) [inline content/gsoc/2024.md:449]: Didn't you mean `RzIL` instead of ESIL?

**Review by Rot127 — APPROVED** [review]

File: content/gsoc/2024.md (hunk 7 above)
Comment (XVilka) [reply content/gsoc/2024.md]: Good catch!

File: content/gsoc/2024.md (hunk 2 above)
Comment (XVilka) [reply content/gsoc/2024.md:326]: Not sure what you are talking about? It has only one.

File: content/gsoc/2024.md (hunk 1 above)
Comment (XVilka) [reply content/gsoc/2024.md:223]: Lets keep this for now, we can remove if/when we are accepted.

Comment (wargio) [reply content/gsoc/2024.md:223]: ok

File: content/gsoc/2024.md [hunk 8]
```diff
@@ -0,0 +1,606 @@
...
+ - thestr4ng3r
+ - ret2libc
+
+### Links/Resources
+- [Debug-labeled issues](https://github.com/rizinorg/rizin/labels/Debug)
+- [RzDebug-labaled issues](https://github.com/rizinorg/rizin/labels/RzDebug)
+- [New Platform support](https://github.com/rizinorg/ideas/labels/New%20Platform)
+- [New Architecture support](https://github.com/rizinorg/ideas/labels/New%20Architecture)
+
+## FRIDA integration (175 hour project)
```
Comment (wargio) [inline content/gsoc/2024.md:374]: @XVilka

File: content/gsoc/2024.md (hunk 2 above)
Comment (wargio) [reply content/gsoc/2024.md:326]:
```suggestion
```

Comment (wargio) [reply content/gsoc/2024.md:326]: i think this was something that was moved further down and this specific section remained.

**Review by wargio — APPROVED** [review]
LGTM besides the double frida thingy

File: content/gsoc/2024.md [hunk 9]
```diff
@@ -0,0 +1,606 @@
...
+projects.
+
+## Tasks
+1. Rewrite C++ demangler to and remove the GPL code
+2. Rewrite some of the mainstream architectures that still rely on binutils without using GPL-only
+   code
+
+Good example of such architectures are:
+- SPARC (there is already capstone-based RzAsm and RzAnalysis plugin but it's less complete than binutils-based one, capstone update is also required)
+- Xtensa (It's better to implemented/updated in capstone)
```
Comment (Rot127) [inline content/gsoc/2024.md:422]:
```suggestion
- Xtensa (It's better to implement/update it in capstone)
```

File: content/gsoc/2024.md [hunk 10]
```diff
@@ -0,0 +1,606 @@
...
+
+The `rz-gg` tool while has the ability to create a custom shellcode has the outdated database of the shellcodes, so [updating them](https://github.com/rizinorg/rizin/issues/154) is crucial for the tool to be relevant.
+
+### Task
+1. Update the shellcodes database, imrove `rz-gg` features and documentation
```
Comment (Rot127) [inline content/gsoc/2024.md:454]:
```suggestion
1. Update the shellcodes database, improve `rz-gg` features and documentation
```

**Review by Rot127 — APPROVED** [review]
Besides the two thinks LGTM.

**Pull request merged**